src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -40,44 +45,40 @@ export function useInterpolator(): D3Interpolator {
   return INTERPOLATORS[colorMap];
 }
 
-export function useDataScale(): DataScale | undefined {
-  const [dataDomain, customDomain, hasLogScale] = useHeatmapConfig(
-    state => [state.dataDomain, state.customDomain, state.hasLogScale],
+export function useTextureData(): Uint8Array | undefined {
+  const [dataDomain, customDomain, hasLogScale, colorMap] = useHeatmapConfig(
+    state => [
+      state.dataDomain,
+      state.customDomain,
+      state.hasLogScale,
+      state.colorMap,
+    ],
     shallow
   );
 
-  return useMemo(() => {
-    if (!dataDomain) {
-      return undefined;
-    }
-
-    const scale = (hasLogScale ? scaleSymlog : scaleLinear)();
-    scale.domain(customDomain || dataDomain);
-
-    return scale;
-  }, [customDomain, dataDomain, hasLogScale]);
-}
-
-export function useTextureData(): Uint8Array | undefined {
-  const dataScale = useDataScale();
-  const interpolator = useInterpolator();
   const values = useValues();
 
-  return useMemo(() => {
-    if (!dataScale) {
-      return undefined;
-    }
+  const { proxy } = useComlink<TextureWorker>(() => new Worker(), []);
+  const [textureData, setTextureData] = useState<Uint8Array>();
 
-    const colorScale = scaleSequential(interpolator);
+  useEffect(() => {
+    if (!dataDomain) {
+      return;
+    }
 
-    // Compute RGB color array for each datapoint `[[<r>, <g>, <b>], [<r>, <g>, <b>], ...]`
-    const colors = values.map(val => {
-      const { r, g, b } = rgb(colorScale(dataScale(val))); // `scale` returns CSS RGB strings
-      return [r, g, b];
-    });
+    (async () => {
+      setTextureData(
+        await proxy.computeTextureData(
+          values,
+          customDomain || dataDomain,
+          hasLogScale,
+          colorMap
+        )
```

Comment:
Thanks to `comlink`, it's just like calling an async method!

Reply:
Sweet !